FirebaseInstallations/Source/Library/InstallationsAPI/FIRInstallationsAPIService.m

```diff
@@ -70,53 +70,93 @@ - (instancetype)initWithURLSession:(NSURLSession *)URLSession
   });
 }
 
+- (FBLPromise<FIRInstallationsItem *> *)refreshAuthTokenForInstallation:
+    (FIRInstallationsItem *)installation {
+  NSURLRequest *request = [self authTokenRequestWithInstallation:installation];
+  return [self sendURLRequest:request]
+      .then(^FBLPromise<FIRInstallationsStoredAuthToken *> *(NSArray *_Nullable value) {
+        return [self authTokenWithServerResponse:value.lastObject responseData:value.firstObject];
+      })
+      .then(^FIRInstallationsItem *(FIRInstallationsStoredAuthToken *authToken) {
```

Comment:
Is the second "then" provide the reject logic?

Reply:
The second `then` will be performed only if the previous one is fulfilled. 
Or maybe I misunderstood the question...